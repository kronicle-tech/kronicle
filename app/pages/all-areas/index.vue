<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllAreasTabs />

          <b-alert show="10" dismissible variant="info" class="my-3">
            Click an area's name in the table below to view more information about
            that area
          </b-alert>

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
              <tr>
                <th class="area-name">Name</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr v-for="area in areas" :key="area.id">
                <td class="area-name table-primary">
                  <AreaName :area="area" />
                </td>
                <td>
                  <Markdown :markdown="area.description" />
                </td>
              </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BAlert, BCol, BContainer, BRow} from 'bootstrap-vue'
import { Area } from '~/types/kronicle-service'
import AllAreasTabs from '~/components/AllAreasTabs.vue'
import AreaName from '~/components/AreaName.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllAreasTabs,
    'b-alert': BAlert,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    AreaName,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id,name,description)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    return {
      areas,
    }
  },
  data() {
    return {
      areas: [] as Area[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Areas',
    }
  },
})
</script>
