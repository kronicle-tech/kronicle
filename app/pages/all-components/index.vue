<template>
  <div>
    <AllComponentsTabs />
    <b-alert show="10" dismissible variant="info" class="my-3">
      Click a component's name in the table below to view more information about
      that component
    </b-alert>

    <ComponentsView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BAlert } from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Component } from '~/types/kronicle-service'
import AllComponentsTabs from '~/components/AllComponentsTabs.vue'
import ComponentsView from '~/components/ComponentsView.vue'

export default Vue.extend({
  components: {
    AllComponentsTabs,
    'b-alert': BAlert,
    ComponentsView,
  },
  async asyncData({ $config, route, store }) {
    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId)`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    store.commit('componentFilters/initialize', {
      components,
      route,
    })

    return {
      components,
    }
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Components',
    }
  },
})
</script>
