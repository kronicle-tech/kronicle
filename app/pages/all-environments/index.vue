<template>
  <div class="m-3">
    <EnvironmentStatesView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {NuxtRuntimeConfig} from "@nuxt/types/config/runtime";
import {Route} from "vue-router";
import {Store} from "vuex";
import EnvironmentStatesView from '~/components/EnvironmentStatesView.vue'
import { Component } from '~/types/kronicle-service'

interface Data {
  components: Component[]
}

async function loadData(
  { $config, route, store }: { $config: NuxtRuntimeConfig, route: Route, store: Store<any> }
): Promise<Data> {
  const components = await fetch(
    `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId,states)`
  )
    .then((res) => res.json())
    .then((json) => json.components as Component[])

  store.commit('componentFilters/initialize', {
    components,
    route,
  })

  return {
    components: [] as Component[],
  }
}

export default Vue.extend({
  components: {
    EnvironmentStatesView,
  },
  async asyncData({ $config, route, store }) {
    return await loadData({
      $config,
      route,
      store,
    })
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Environments',
    }
  },
  mounted() {
    setInterval(
      async () => {
        const data = await loadData({
          $config: this.$config,
          route: this.$route,
          store: this.$store
        })
        this.components = data.components
      },
      60 * 1000
    )
  },
})
</script>
