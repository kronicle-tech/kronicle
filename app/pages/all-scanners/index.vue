<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllScannersTabs />

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
            <tr>
              <th class="scanner-id">Id</th>
              <th class="scanner-description">Description</th>
              <th class="scanner-notes">Notes</th>
            </tr>
            </thead>
            <tbody>
            <tr v-for="scanner in scanners" :key="scanner.id">
              <td class="scanner-id">
                {{ scanner.id }}
              </td>
              <td class="scanner-description">
                <Markdown :markdown="scanner.description" />
              </td>
              <td class="scanner-notes">
                <Markdown :markdown="scanner.notes" />
              </td>
            </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BCol, BContainer, BRow} from "bootstrap-vue";
import { Scanner } from '~/types/kronicle-service'
import AllScannersTabs from '~/components/AllScannersTabs.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllScannersTabs,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    Markdown,
  },
  async asyncData({ $config }) {
    const scanners = await fetch(
      `${$config.serviceBaseUrl}/v1/scanners?fields=scanners(id,description,notes)`
    )
      .then((res) => res.json())
      .then((json) => json.scanners as Scanner[])

    return {
      scanners,
    }
  },
  data() {
    return {
      scanners: [] as Scanner[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Scanners',
    }
  },
})
</script>
