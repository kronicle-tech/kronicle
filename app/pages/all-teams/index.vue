<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllTeamsTabs />

          <b-alert show="10" dismissible variant="info" class="my-3">
            Click a team's name in the table below to view more information about that
            team
          </b-alert>

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
              <tr>
                <th class="team-name">Name</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr v-for="team in teams" :key="team.id">
                <td class="team-name table-primary">
                  <TeamName :team="team" />
                </td>
                <td>
                  <Markdown :markdown="team.description" />
                </td>
              </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BAlert, BCol, BContainer, BRow} from 'bootstrap-vue'
import { Team } from '~/types/kronicle-service'
import AllTeamsTabs from '~/components/AllTeamsTabs.vue'
import TeamName from '~/components/TeamName.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllTeamsTabs,
    'b-alert': BAlert,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    TeamName,
    Markdown,
  },
  async asyncData({ $config }) {
    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id,name,emailAddress,description)`
    )
      .then((res) => res.json())
      .then((json) => json.teams as Team[])

    return {
      teams,
    }
  },
  data() {
    return {
      teams: [] as Team[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Teams',
    }
  },
})
</script>
