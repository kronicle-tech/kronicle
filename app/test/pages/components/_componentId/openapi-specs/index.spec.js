import Index from '@/pages/components/_componentId/openapi-specs/index.vue'
import { createPageWrapper } from '~/test/pages/pageUtils'
import {
  createComponent,
  createComponentWithOpenApiSpecs,
} from '~/test/testDataUtils'

describe('Index', () => {
  const route = {
    params: {
      componentId: 'test-component-id-1',
    },
  }
  let component
  let wrapper
  async function createWrapper() {
    wrapper = await createPageWrapper(Index, {
      route,
      serviceRequests: {
        '/v1/components/test-component-id-1?fields=component(id,name,teams,openApiSpecs)':
          {
            responseBody: { component },
          },
      },
    })
  }

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when Get Component service endpoint returns a component', () => {
    beforeEach(() => {
      component = createComponent({ componentNumber: 1 })
    })

    test('has the right page title', async () => {
      await createWrapper()
      expect(wrapper.vm.$metaInfo.title).toBe(
        'Kronicle - Test Component Name 1 - OpenAPI Specs'
      )
    })

    describe('when the component has no OpenAPI specs', () => {
      test('renders no OpenAPI specs', async () => {
        await createWrapper()
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('when the component has OpenAPI specs', () => {
      beforeEach(() => {
        component = createComponentWithOpenApiSpecs({ componentNumber: 1 })
      })

      test('renders the details of the OpenAPI specs', async () => {
        await createWrapper()
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
