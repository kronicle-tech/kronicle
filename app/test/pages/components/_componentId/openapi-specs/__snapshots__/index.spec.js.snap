// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index when Get Component service endpoint returns a component when the component has OpenAPI specs renders the details of the OpenAPI specs 1`] = `
<div
  class="m-3"
>
  <h1
    class="text-info"
  >
    Test Component Name 1 - OpenAPI Specs
  </h1>
   
  <div
    class="my-3"
  >
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1"
      target="_self"
    >
      Info
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/dependencies"
      target="_self"
    >
      
    Visualizations
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/response-times"
      target="_self"
    >
      
    Response Times
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/downstream-response-times"
      target="_self"
    >
      
    Downstream Response Times
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/call-graphs"
      target="_self"
    >
      
    Call Graphs
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/cross-functional-requirements"
      target="_self"
    >
      
    Cross Functional Requirements
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tech-debts"
      target="_self"
    >
      
    Tech Debts
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tests"
      target="_self"
    >
      Tests
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/errors"
      target="_self"
    >
      Errors
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/openapi-specs"
      target="_self"
    >
      
    OpenAPI
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/graphql-schemas"
      target="_self"
    >
      
    GraphQL
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/repo"
      target="_self"
    >
      Repo
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/readme"
      target="_self"
    >
      README
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/to-dos"
      target="_self"
    >
      To Dos
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/lines-of-code"
      target="_self"
    >
      
    Lines of Code
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/software-repositories"
      target="_self"
    >
      
    Software Repositories
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/software"
      target="_self"
    >
      Software
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/imports"
      target="_self"
    >
      Imports
    </a>
  </div>
   
  <div>
    <div
      class="card"
    >
      <!---->
      <!---->
      <div
        class="list-group"
      >
        <div
          class="list-group-item list-group-item-success"
        >
          <span
            class=""
          >
            2
          </span>
          
        OpenAPI specs
      
        </div>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="mt-3 mb-4"
      components="[object Object]"
    >
       
      <button
        aria-controls="filters"
        aria-expanded="false"
        class="btn btn-primary collapsed"
        id="toggleFilters"
        type="button"
      >
        <svg
          aria-hidden="true"
          aria-label="blank"
          class="bi-blank b-icon bi"
          fill="currentColor"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g />
        </svg>
         Filters
  
      </button>
       
      <transition-stub
        class="mt-3"
        css="true"
        enteractiveclass="collapsing"
        enterclass=""
        entertoclass="collapse show"
        leaveactiveclass="collapsing"
        leaveclass="collapse show"
        leavetoclass="collapse"
      >
        <div
          class="collapse"
          id="filters"
          style="display: none;"
        >
          <div
            class="card-columns"
          >
             
            <!---->
             
            <!---->
             
            <!---->
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__117"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__117__BV_label_"
                    tabindex="-1"
                  >
                    Teams
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__118"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__118_BV_option_0"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__118_BV_option_0"
                        >
                          <span>
                            test-team-id-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__118_BV_option_1"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__118_BV_option_1"
                        >
                          <span>
                            test-team-id-1-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__121"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__121__BV_label_"
                    tabindex="-1"
                  >
                    Tags
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__122"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__122_BV_option_0"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__122_BV_option_0"
                        >
                          <span>
                            test-tag-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__122_BV_option_1"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__122_BV_option_1"
                        >
                          <span>
                            test-tag-1-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__125"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__125__BV_label_"
                    tabindex="-1"
                  >
                    Component Types
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__126"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__126_BV_option_0"
                          name="componentTypeId"
                          type="checkbox"
                          value="test-component-type-id-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__126_BV_option_0"
                        >
                          <span>
                            test-component-type-id-1
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__128"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__128__BV_label_"
                    tabindex="-1"
                  >
                    Platforms
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__129"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__129_BV_option_0"
                          name="platformId"
                          type="checkbox"
                          value="undefined"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__129_BV_option_0"
                        >
                          <span>
                            undefined
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <div
                  class="form-group"
                  id="__BVID__131"
                  role="group"
                >
                  <label
                    class="d-block"
                    for="component-filter"
                    id="__BVID__131__BV_label_"
                  >
                    Component
                  </label>
                  <div>
                    <select
                      class="custom-select custom-select-sm"
                      id="component-filter"
                    >
                      <option
                        value=""
                      >
                        Please select a component
                      </option>
                      <option
                        value="test-component-id-1"
                      >
                        test-component-id-1
                      </option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
     
    <table
      class="table table-dark"
    >
      <thead>
        <tr>
          <th
            class="component"
          >
            Component
          </th>
           
          <th
            class="teams"
          >
            Teams
          </th>
           
          <th
            class="location"
          >
            Location
          </th>
           
          <th
            class="action"
          >
            Action
          </th>
           
          <th
            class="description"
          >
            Description
          </th>
        </tr>
      </thead>
       
      <tbody>
        <tr>
          <td
            class="component"
          >
            <a
              href="/components/test-component-id-1"
            >
              Test Component Name 1
            </a>
          </td>
           
          <td
            class="teams"
          >
            <ul>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-1-1"
                >
                  test-team-id-1-1
                </a>
                 
                <!---->
                 
                <!---->
              </li>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-1-2"
                >
                  test-team-id-1-2
                </a>
                 
                <!---->
                 
                <!---->
              </li>
            </ul>
          </td>
           
          <td
            class="location"
          >
            
        https://example.com/test-openapi-spec-1-1
      
          </td>
           
          <td
            class="action table-secondary"
          >
            <a
              class="btn btn-info"
              href="/components/test-component-id-1/openapi-specs/1"
              target="_self"
            >
              
          View OpenAPI spec
        
            </a>
          </td>
           
          <td
            class="description"
          >
            <div>
              

              <p>
                Test OpenAPI Spec Description 1 1
              </p>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="component"
          >
            <a
              href="/components/test-component-id-1"
            >
              Test Component Name 1
            </a>
          </td>
           
          <td
            class="teams"
          >
            <ul>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-1-1"
                >
                  test-team-id-1-1
                </a>
                 
                <!---->
                 
                <!---->
              </li>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-1-2"
                >
                  test-team-id-1-2
                </a>
                 
                <!---->
                 
                <!---->
              </li>
            </ul>
          </td>
           
          <td
            class="location"
          >
            
        https://example.com/test-openapi-spec-1-2
      
          </td>
           
          <td
            class="action table-secondary"
          >
            <a
              class="btn btn-info"
              href="/components/test-component-id-1/openapi-specs/2"
              target="_self"
            >
              
          View OpenAPI spec
        
            </a>
          </td>
           
          <td
            class="description"
          >
            <div>
              

              <p>
                Test OpenAPI Spec Description 1 2
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Index when Get Component service endpoint returns a component when the component has no OpenAPI specs renders no OpenAPI specs 1`] = `
<div
  class="m-3"
>
  <h1
    class="text-info"
  >
    Test Component Name 1 - OpenAPI Specs
  </h1>
   
  <div
    class="my-3"
  >
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1"
      target="_self"
    >
      Info
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/dependencies"
      target="_self"
    >
      
    Visualizations
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/response-times"
      target="_self"
    >
      
    Response Times
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/downstream-response-times"
      target="_self"
    >
      
    Downstream Response Times
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/call-graphs"
      target="_self"
    >
      
    Call Graphs
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/cross-functional-requirements"
      target="_self"
    >
      
    Cross Functional Requirements
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tech-debts"
      target="_self"
    >
      
    Tech Debts
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tests"
      target="_self"
    >
      Tests
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/errors"
      target="_self"
    >
      Errors
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/openapi-specs"
      target="_self"
    >
      
    OpenAPI
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/graphql-schemas"
      target="_self"
    >
      
    GraphQL
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/repo"
      target="_self"
    >
      Repo
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/readme"
      target="_self"
    >
      README
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/to-dos"
      target="_self"
    >
      To Dos
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/lines-of-code"
      target="_self"
    >
      
    Lines of Code
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/software-repositories"
      target="_self"
    >
      
    Software Repositories
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/software"
      target="_self"
    >
      Software
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/imports"
      target="_self"
    >
      Imports
    </a>
  </div>
   
  <div>
    <div
      class="card"
    >
      <!---->
      <!---->
      <div
        class="list-group"
      >
        <div
          class="list-group-item list-group-item-danger"
        >
          <span
            class="display-1"
          >
            0
          </span>
          
        OpenAPI specs
      
        </div>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="mt-3 mb-4"
      components="[object Object]"
    >
       
      <button
        aria-controls="filters"
        aria-expanded="false"
        class="btn btn-primary collapsed"
        id="toggleFilters"
        type="button"
      >
        <svg
          aria-hidden="true"
          aria-label="blank"
          class="bi-blank b-icon bi"
          fill="currentColor"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g />
        </svg>
         Filters
  
      </button>
       
      <transition-stub
        class="mt-3"
        css="true"
        enteractiveclass="collapsing"
        enterclass=""
        entertoclass="collapse show"
        leaveactiveclass="collapsing"
        leaveclass="collapse show"
        leavetoclass="collapse"
      >
        <div
          class="collapse"
          id="filters"
          style="display: none;"
        >
          <div
            class="card-columns"
          >
             
            <!---->
             
            <!---->
             
            <!---->
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__73"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__73__BV_label_"
                    tabindex="-1"
                  >
                    Teams
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__74"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__74_BV_option_0"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__74_BV_option_0"
                        >
                          <span>
                            test-team-id-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__74_BV_option_1"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__74_BV_option_1"
                        >
                          <span>
                            test-team-id-1-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__77"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__77__BV_label_"
                    tabindex="-1"
                  >
                    Tags
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__78"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__78_BV_option_0"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__78_BV_option_0"
                        >
                          <span>
                            test-tag-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__78_BV_option_1"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__78_BV_option_1"
                        >
                          <span>
                            test-tag-1-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__81"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__81__BV_label_"
                    tabindex="-1"
                  >
                    Component Types
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__82"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__82_BV_option_0"
                          name="componentTypeId"
                          type="checkbox"
                          value="test-component-type-id-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__82_BV_option_0"
                        >
                          <span>
                            test-component-type-id-1
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__84"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__84__BV_label_"
                    tabindex="-1"
                  >
                    Platforms
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__85"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__85_BV_option_0"
                          name="platformId"
                          type="checkbox"
                          value="undefined"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__85_BV_option_0"
                        >
                          <span>
                            undefined
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body"
              >
                <!---->
                <!---->
                <div
                  class="form-group"
                  id="__BVID__87"
                  role="group"
                >
                  <label
                    class="d-block"
                    for="component-filter"
                    id="__BVID__87__BV_label_"
                  >
                    Component
                  </label>
                  <div>
                    <select
                      class="custom-select custom-select-sm"
                      id="component-filter"
                    >
                      <option
                        value=""
                      >
                        Please select a component
                      </option>
                      <option
                        value="test-component-id-1"
                      >
                        test-component-id-1
                      </option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
     
    <!---->
  </div>
</div>
`;
